examples/webview2/src/main.rs

```diff
@@ -0,0 +1,622 @@
+
+        if webview.frame.is_some() {
+            WebView::set_window_webview(parent, Box::new(webview.clone()));
+        }
+
+        Ok(webview)
+    }
+
+    pub fn run(self) -> Result<()> {
+        let webview = self.webview.as_ref();
+        let url = self.url.lock().expect("lock url").clone();
+        let (tx, rx) = mpsc::channel();
+
+        if url.len() > 0 {
+            let token =
+                webview.NavigationCompleted(TypedEventHandler::<
+                    CoreWebView2,
+                    CoreWebView2NavigationCompletedEventArgs,
+                >::new(move |_sender, _args| {
+                    tx.send(()).expect("send over mpsc channel");
+                    Ok(())
+                }))?;
+
+            webview.Navigate(&url)?;
+
+            let result = wait_with_pump(rx);
+            webview.RemoveNavigationCompleted(token)?;
+            result?;
+        }
+
+        if let Some(frame) = self.frame.as_ref() {
+            let hwnd = *frame.window;
+            unsafe {
+                WindowsAndMessaging::ShowWindow(hwnd, SHOW_WINDOW_CMD::SW_SHOW);
+                Gdi::UpdateWindow(hwnd);
+                KeyboardAndMouseInput::SetFocus(hwnd);
+            }
+        }
+
+        let mut msg = MSG::default();
+        let h_wnd = HWND::default();
+
+        loop {
+            while let Ok(f) = self.rx.try_recv() {
+                (f)(self.clone());
+            }
+
+            unsafe {
+                let result = WindowsAndMessaging::GetMessageA(&mut msg, h_wnd, 0, 0).0;
+
+                match result {
+                    -1 => {
+                        break {
+                            let error =
+                                format!("GetMessageW failed: {}", Win32::Debug::GetLastError());
+                            Err(Error::new(ErrorCode::E_NOINTERFACE, &error))
+                        }
+                    }
+                    0 => break Ok(()),
+                    _ => match msg.message {
+                        WindowsAndMessaging::WM_APP => (),
+                        _ => {
+                            WindowsAndMessaging::TranslateMessage(&msg);
+                            WindowsAndMessaging::DispatchMessageA(&msg);
+                        }
+                    },
+                }
+            }
+        }
+    }
+
+    pub fn terminate(self) -> Result<()> {
+        self.dispatch(|_webview| unsafe {
+            WindowsAndMessaging::PostQuitMessage(0);
+        })?;
+        Ok(())
+    }
+
+    pub fn set_title(&self, title: &str) -> Result<&Self> {
+        if let Some(frame) = self.frame.as_ref() {
+            if let Ok(title) = CString::new(title) {
+                unsafe {
+                    WindowsAndMessaging::SetWindowTextA(
+                        *frame.window,
+                        PSTR(title.as_ptr() as *mut _),
+                    );
+                }
+            }
+        }
```

Comment:
```suggestion
            unsafe {
                WindowsAndMessaging::SetWindowTextA(*frame.window, title);
            }
```

The Windows crate provides null termination automatically for functions like `SetWindowTextA`.

Reply:
Interesting, but not for `WNDCLASSA`, so I still need to use `CString` there it looks like. Even though it I have a copy already null terminated there for `WNDCLASSA`, I decided it's safer to let the `CreateWindowExA` call re-terminate the &str literal instead of sharing the `CString` pointer.

Reply:
Yes, the `WNDCLASSA` has a `PSTR` field so I can't as easily inject the conversion as I do for input parameters (https://github.com/microsoft/windows-rs/issues/645#issuecomment-808372812). There's also a desire to add a `const` function to generate a `PSTR` at compile time that would make that cleaner (#581).